libs/vkd3d/memory.c

```diff
@@ -492,6 +492,18 @@ static void vkd3d_memory_transfer_queue_clear_allocation(struct vkd3d_memory_tra
         /* Probably faster than doing this on the GPU
          * and having to worry about synchronization */
         memset(allocation->cpu_address, 0, allocation->resource.size);
+
+        if (allocation->flags & VKD3D_ALLOCATION_FLAG_HOST_NON_COHERENT)
+        {
+            struct d3d12_device *device = queue->device;
+            const struct vkd3d_vk_device_procs *vk_procs = &device->vk_procs;
+            VkMappedMemoryRange mapped_range = { 0 };
+            mapped_range.memory = allocation->device_allocation.vk_memory;
+            mapped_range.offset = allocation->offset;
+            mapped_range.size = allocation->resource.size;
+
+            VK_CALL(vkFlushMappedMemoryRanges(device->vk_device, 1, &mapped_range));
```

Comment:
Might make more sense to just unconditionally call FlushMappedMemoryRanges. It seems like we do that for ID3D12Resource::Map/Unmap.

However, I think we're likely missing flushes in way more places going by this fix ...

Reply:
Yeah, I'm afraid of that too.

I'll scout around `->cpu_address` accesses.

Reply:
Added vkd3d_format_copy_data locations too.